server/utils/match/occupancySummary.ts

```diff
@@ -0,0 +1,76 @@
+import type { Cas1PremiseCapacity, Cas1PremiseCapacityForDay } from '@approved-premises/api'
+import { addDays, differenceInDays } from 'date-fns'
+import { DateFormats, daysToWeeksAndDays } from '../dateUtils'
+
+export type DateRange = {
+  start: string
+  end: string
+}
+
+const daysToRanges = (days: Array<Cas1PremiseCapacityForDay>) =>
+  days.reduce((ranges: Array<DateRange>, capacityForDay) => {
+    if (!ranges.length) {
+      ranges.push({ start: capacityForDay.date, end: capacityForDay.date })
+    } else {
+      const lastRange = ranges[ranges.length - 1]
+      if (differenceInDays(capacityForDay.date, lastRange.end) === 1) {
+        lastRange.end = capacityForDay.date
+      } else {
+        ranges.push({ start: capacityForDay.date, end: capacityForDay.date })
+      }
+    }
+    return ranges
+  }, [])
+
+export const renderDateRange = (dateRange: DateRange): string => {
+  let render = `${DateFormats.isoDateToUIDate(dateRange.start)}`
+  const totalDays = DateFormats.differenceInDays(
```

Comment:
can we not use the same `differenceInDays` function as on line 16? If not would reduce the need to import `DateFormats`

Reply:
Yes, that's right actually -- I imported this one before I implemented the week/days format, which means we can now use the other one!

Reply:
Addressed as suggested -- `DateFormats` is imported anyway, but this does clean up the syntax a bit. 

Also refactored the calculation of the difference slightly: now simply adding `1` to the difference, rather than checking difference between `addDays(endDate, 1)` and `startDate` -- I think this is more readable now, and removes the need for `addDays` to be imported at all.

Reply:
ah yes, of course DateFormats is used for other stuff. Cool, all sounds good